src/configuration/app/source-operations.md

```diff
@@ -0,0 +1,74 @@
+---
+search:
+    keywords: ['.platform.app.yaml', 'automated', 'code', 'source', 'update', 'operation', 'dependencies', 'auto']
+---
+
+# Source operations
+
+> **note**
+>
+> Source Operations are currently in beta.  While the syntax is not expected to change some behavior might in the future.
+
+An application can define a number of operations that apply to its source code and that can be automated.
+
+A basic, common source operation could be to automatically update Composer dependencies like this:
+
+```yaml
+source:
+    operations:
+        update:
+            command: |
+                set -e
+                composer update
+                git add composer.lock
+                git commit -m "Update Composer dependencies."
+```
+
+The `update` key is the name of the operation. It is arbitrary, and multiple source operations can be defined.
+
+(You may with to include more robust error handling than this example.)
+
+The environment resource gets a new `source-operation` action which can be triggered by the CLI:
+
+```bash
+platform source-operation:run update
+```
+
+The `source-operation:run` command takes the command name to run plus, optionally, additional variables to inject in the environment of the source operation.  They will be interpreted the same way as any other [variable](/development/variables.md) set through the UI or CLI, which means you need an `env:` prefix to expose them as a Unix environment variable.  They can then be referenced by the source operation like any other variable.
```

Comment:
command takes name of the command name to run. Additional variables can be added to inject into the environment of the source operation.